File: java-guessing-game/guessTheNumber/src/guessTheNumber.java
import java.util.Scanner;
import java.util.concurrent.ThreadLocalRandom;

public class guessTheNumber {
    public static void main(String[] args){
        Scanner input = new Scanner(System.in);
        String playAgain = "yes";
        int score = 0;

        //STARTING THE MAIN WHILE LOOP
        while (playAgain.equalsIgnoreCase("yes")) {
            //CREATING A RANDOM NUMBER
            int randomNumber = ThreadLocalRandom.current().nextInt(1, 21);
            int attempts = 5;

            //CREATING AN INNER LOOP
            while (attempts > 0) {
                System.out.print("🎲 Choose your Random Number(1-20): ");
                String guess = input.nextLine();
                int guessNumber = 0;

                //ENSURING AN INTEGER IS INPUTED
                try{
                    guessNumber = Integer.parseInt(guess);
                } catch (NumberFormatException e){
                    System.out.println("‼️ Invalid Input, Please Try Again!!");
                }

                if(guessNumber == randomNumber){
                    score++;
                    System.out.println("🧩 HAAZZZAAAH, You win! Current Score: " + score);
                    break;
                } else if(guessNumber > randomNumber){
                    attempts--;
                    System.out.println("🙂‍↔️Oops Too High mei old chap!!! You have " + attempts + " left");
                } else if(guessNumber < randomNumber){
                    attempts--;
                    System.out.println("👾 Oops Too low mei old chap!!! You have " + attempts + " left");
                }


                //CLOSING THE INNER LOOP
                if(attempts == 0){
                    System.out.println("🎰 Oops seems like you're out of Attempts friend The Correct number was: " + randomNumber + ", Try replaying the Game 🐝!");
                }
            }


           //CLOSING THE MAIN LOOP
            System.out.print("Would You like to play Another Round 🎮(yes/no): ");
            input.nextLine();
            playAgain = input.nextLine();
        }

        System.out.println("==== 👾 Thank you for Trying My Mini Game 🎰 =====");
        input.close();
    }
}

File: guessing-game-python/guessTheNumber.py
#IMPORTING THREADING & RANDOM
import threading
import random

#WELCOME MESSAGE
print("===== 🎲 Mini Number Guessing Game 🎮 ====")

#MAIN FUNCTION

def guess_the_number():
    score = 0

    #STARTING THE MAIN WHILE LOOP
    while True:

        #CREATING THE RANDOM NUMBER
        random_number = random.randint(1, 20)
        attempts = 5

        #GET THE USER INPUT
        user_guess = input("🎰 Choose a Random Number (1-20): ")

        #STARTING THE INNER LOOP TO RETRY IF OUT OF TRIES
        while True:

            #FIRST EVALUATE IF THE USER INPUTED A NUMBER
            try:
                guess = int(user_guess)
            except ValueError:
                print("🚫 Invalid Input, please ensure ypu input a number")

            if guess == random_number:
                print(f"HAZZAH🎰 You guessed {guess} and the correct number was {random_number} 👾 You Win🎊")
                break
            elif guess > random_number:
                attempts -= 1
                print(f"Ooops 🐝 Too {guess} is too High, You have {attempts} attempts left ‼️")
            elif guess < random_number:
                attempts -= 1
                print(f"Ooops 🐝 Too {guess} is too Low, You have {attempts} attempts left ‼️")
            


        #CLOSING THE MAIN LOOP
        playAgain = input("🧩 Wanna give it another try?(yes/no): ").lower()
        if playAgain != "yes":
            print("==== 🎮 Thank You Trying My Mini Game 👾 ====")
            break


#TIMEOUT FUNCTION
timer = threading.Timer(2, guess_the_number)
timer.start()